cmd/kubeadm/app/phases/upgrade/compute_test.go

```diff
@@ -221,7 +182,45 @@ func TestGetAvailableUpgrades(t *testing.T) {
 			},
 			allowExperimental: false,
 			errExpected:       false,
-			etcdClient:        etcdClient,
+		},
+		{
+			name: "simple patch version upgrade with external etcd",
```

Comment:
i think we should also have a unit test for the `-0 -> -x` case we are trying to resolve.

Reply:
The problem here is that it's pretty difficult to have such unit test. It should part of testing `performEtcdStaticPodUpgrade`. However, that function has no unit test and it would be difficult to write a meaningful test. It mainly consists of business logic and we have to mock basically everything to be able to check that logic.

Reply:
one potential solution, is to extract just the version comparisons between desired and current versions into a separate function and unit tests it for some special cases.

then again this will be testing the version util at this point which is not needed as we know that `-1` > `-0`.